src/dataset.jl

```diff
@@ -255,8 +257,36 @@ function writelayers(
     dataset::AbstractDataset,
     filename::AbstractString;
     driver::Driver = getdriver(dataset),
-    options = [""],
-    layer_options = [""],
+    options = String[],
+    layers = 0:(nlayer(dataset)-1),
+    layer_options = String[],
+    chunksize::Integer = 20000,
+    use_gdal_copy::Bool = true,
+)
+    create(filename; driver = driver, options = options) do target
+        return copylayers!(
+            dataset,
+            target;
+            layers = layers,
+            layer_options = layer_options,
+            chunksize = chunksize,
+            use_gdal_copy = use_gdal_copy,
+        )
+    end
+    return nothing
+end
+
+"""
+    copylayers(source_dataset, target_dataset; kwargs...)
+
+Copies a (sub)set of layers from source to target dataset. The keyword arguments are the same as for `writelayers`.
+Note that eventual layer options have to conform with the target dataset driver.
+"""
+function copylayers!(
```

Comment:
In the spirit of factoring things out -- should we make this `copylayer!(source_dataset, target_dataset, source_layer_index)` and letting the user write a loop over the layers they want to copy?

Reply:
I think that would not give the user any benefit, so I'd keep it here.